Pal 131 move documentation to new repo 2nd draft (#484)
gchq/Palisade: README.md

limitations under the License.
--->

# <img src="logos/logo.svg" width="180">
### A Tool for Complex and Scalable Data Access Policy Enforcement

Palisade aims to provide a stable framework for defining and enforcing policies for accessing data.

> #### Introduction

Author: Put the bullet points in.

> #### Introduction

Author: Replaced "this" with "any".  Now it reads "to be sanitised for any particular query"

> #### Introduction

Author: Added the "to" .

> #### Introduction

Author: Re-inserted changes to this paragraph.  Should be a better read.

> #### Introduction

Author: Re-inserted changes to this paragraph.  Should be a better read.

limitations under the License.
--->

# <img src="logos/logo.svg" width="180">
### A Tool for Complex and Scalable Data Access Policy Enforcement

Palisade aims to provide a stable framework for defining and enforcing policies for accessing data.

#### Introduction

> Modern organisations can have multiple datasets that are each managed and accessed according to different policies and procedures. This can make it complicated for analysis while still ensuring regulatory compliance.  Most technologies "out of the box" do not provide the necessary solution for managing these complex issues for data access. This leads to access to data being overly restrictive with potential insights from data being lost.   

Reviewer: As these are markdown files, the general consensus is that it should stylistically be presented inside the file as one sentence per line.
See [this thread](https://github.com/prettier/prettier/issues/3302) discussing pros and cons, the biggest here being git diffs work per line.

> Palisade can provide a solution to this problem.  It consists of a set of services for managing and enforcing complex data access policies across different datasets and platforms.  These services are able to provide modular data access controllers and independent peripheral services to allow policies to be integral to the process a client uses to access data.  

Reviewer: Being super picky, this should be "Palisade can provide a solution to this problem; it consists..."

> Consider the example of a set of datasets may have been assembled for specific purposes.   For a given dataset, only particular types of query are permitted for certain types of user.  Also, within each dataset a specific user may only be permitted to access a particular subset of the records. Palisade allows the policies for accessing datasets, redacting records and executing the query to be specified independently of the physical data and the platform hosting the data.  This will enable the clients to perform queries across the datasets while still conforming to company and regulatory policies.  

Reviewer: First -> Second sentence is triple-spaced, but see comment on one-sentence-per-line

> Consider the example of a set of datasets may have been assembled for specific purposes.   For a given dataset, only particular types of query are permitted for certain types of user.  Also, within each dataset a specific user may only be permitted to access a particular subset of the records. Palisade allows the policies for accessing datasets, redacting records and executing the query to be specified independently of the physical data and the platform hosting the data.  This will enable the clients to perform queries across the datasets while still conforming to company and regulatory policies.  

Reviewer: Second sentence, "particular types of queries"

> Policies are set per resource and are not changed by the user or context.  However, the policies can use the information about the user and context to decide how the data needs to be sanitised for any particular query.  This can be applied on a per field basis. This means that fields in a record can be redacted or masked based on: 

Reviewer: 3rd sentence, hyphenate "per-field"

* the contents of the record
* the user accessing the record 

> * the contextual information given by the user or the system at the time of the query.

Reviewer 2: ```suggestion
* the contextual information provided by the user or the system at the time of the query.
```

> Palisade's aim is to provide a stable framework for enforcing policy and auditing requirements for data access requests, while, at the same time, not limiting the solution to any single platform ecosystem. Palisade can be deployed into many different ecosystems such as Hadoop, Kubernetes, AWS, Azure and Google Cloud. As adoption of Palisade grows then burden of creating new connectors between Palisade and new technologies for storing and processing data can be shared across the community.

Reviewer 2: ```suggestion
Palisade's aim is to provide a stable framework for enforcing policy and auditing requirements for data access requests, while, at the same time, not limiting the solution to any single platform ecosystem. Palisade can be deployed into many different ecosystems such as Hadoop, Kubernetes, AWS, Azure and Google Cloud. As adoption of Palisade grows, then the burden of creating new connectors (between Palisade and new technologies for storing and processing data) can be shared across the community.
```

> 

Reviewer: This weird diff is why one-per-line is preferred

> Palisade's aim is to provide a stable framework for defining and enforcing policy and auditing requirements for data access requests, while ,at the same time, not limiting the solution to any single platform ecosystem. Palisade can be adapted into existing solutions that already work in the different ecosystems such as Hadoop, Kubernetes, AWS, Azure and Google Cloud.  This is achieved by sharing with the community the tasks of creating connectors between Palisade and new technologies and solutions for storing data and for processing data.

Reviewer 2: ```suggestion
Palisade's aim is to provide a stable framework for enforcing policy and auditing requirements for data access requests, while, at the same time, not limiting the solution to any single platform ecosystem. Palisade can be deployed into many different ecosystems such as Hadoop, Kubernetes, AWS, Azure and Google Cloud. As adoption of Palisade grows then burden of creating new connectors between Palisade and new technologies for storing and processing data can be shared across the community.
```

# <img src="logos/logo.svg" width="180">

> ###A Tool for Complex and Scalable Data Access Policy Enforcement

Reviewer: Needs a space

Author: Added the space

Palisade aims to provide a stable framework for defining and enforcing policies for accessing data.

> ####Introduction

Reviewer: Needs a space between #### and Introduction... see github render of this markdown

Author: Added the space.

* the contents of the record 
* the user accessing the record 
* and the contextual information given by the user or the system at the time of the query

Palisade's aim is to provide a stable framework for defining and enforcing policy and auditing requirements for data access requests, while ,at the same time, not limiting the solution to any single platform ecosystem. Palisade can  be adapted into existing solutions that already work in the different ecosystems such as Hadoop, Kubernetes, AWS, Azure and Google Cloud.  This is achieved by sharing with the community the tasks of creating connectors between Palisade and new technologies and solutions for storing data and for processing data.

![Palisade Overview](img/Palisade_overview.png)

If required, Palisade allows an organisation to use centralised services which each deployment of Palisade's data access services can share. This means that your data access policies, audit logs, user account details etc. don't need to be duplicated on every system. Of course, Palisade can still be deployed locally on a per platform basis.

> ###Additional Information 

Reviewer: Needs a space

Author: Thanks,  missed that

Author: Added the space

> If required, Palisade allows an organisation to use centralised services which each deployment of Palisade's data access services can share. This means that your data access policies, audit logs, user account details etc. don't need to be duplicated on every system. Of course, Palisade can still be deployed locally on a per platform basis.

Reviewer: last sentence hyphenate "per-platform"
